airflow/bin/cli.py

```diff
@@ -424,15 +427,76 @@ def clear(args):
         confirm_prompt=not args.no_confirm,
         include_subdags=not args.exclude_subdags)
 
+def command_of(pid):
+    """
+    Returns the command (ie argv[0]) of the process with process id @pid
+    """
+    return subprocess.check_output([
+        'ps', str(pid), '-o', 'command'
+    ]).split('\n')[1]
+
+def restart_workers(gunicorn_master_proc, num_workers_expected):
+    """
+    Runs forever, monitoring the child processes of @gunicorn_master_proc
+    and restarting workers occasionally.
+
+    Each iteration of the loop traverses one edge of this state transition
+    diagram, where each state (node) represents
+    [ num_ready_workers_running / num_workers_running ]. We expect most time
+    to be spent in [n / n]. The horizontal transition at ? happens after the
+    new worker parses all the dags.
+
+       V ---------------------------------------------------------------|
+    [n / n] --------> [n / n + 1]  ---------?----> [n + 1 / n + 1] ---- |
+                           ^----------------|
+
+    We change the number of workers by sending TTIN and TTOU to the gunicorn
+    master process, which increases and decreases the number of child workers
+    respectively. Gunicorn guarantees that on TTOU workers are terminated
+    gracefully and that the oldest worker is terminated.
+    """
```

Comment:
@detroitcoder IIRC passing SIGHUP to gunicorn will make it unable to serve requests while dag parsing is occuring. So if your dag bag takes 10 minutes to parse, the web interface will be down for 10 minutes.